src/config.cpp

```diff
@@ -88,10 +206,26 @@ void parse_env() {
     }
 }
 
+void gen_config_file() {
+    std::unordered_map<std::string, std::string> file_config_values;
+    std::ifstream config_def("config.def");
+    std::ofstream config_file("clvk.conf");
+    if (!config_def.is_open()) {
+        std::cerr << "Error opening config.def\n";
+        return;
+    }
+    if (!config_file.is_open()) {
+        std::cerr << "Error opening clvk.conf\n";
+        return;
+    }
+    read_config_file(file_config_values, config_def);
+}
+
 } // namespace
 
 void init_config() {
-    // TODO Parse config file
-    // Parse environment
+    // make a default config file from config.def
```

Comment:
Why is that needed?

Reply:
kevin asked to have a default .conf file generated from the .def file

Reply:
I am not sure he meant that exactly. It feels weird to generate it at run time.